### PR TITLE
DM-33469: Add python=3.10 dev environment

### PR DESCRIPTION
**Pull request by erykoff**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

To make this solve, move cfitsio pin to 3.10 and numpy to 1.21.  Also try to enable the linux_aarch64 build.

### CODE REVIEW

**Review by ktlim — APPROVED** [review]

**Comment by erykoff** [comment]
Looks like `linux_aarch64` isn't solving https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=450985&view=logs&j=6494c7a6-a019-59a3-6f3d-9bab8ca96717

So I guess that this is not happening today, since I don't know how to debug that.

**Comment by erykoff** [comment]
This is ready!

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: conda-forge.yml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 conda_forge_output_validation: true
 build_platform:
   osx_arm64: osx_64
-test_on_native_only: true
+  linux_aarch64: linux_64
```
Comment (beckermr) [inline conda-forge.yml:4]:
```suggestion
```

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: conda-forge.yml [hunk 2]
```diff
@@ -1,4 +1,6 @@
 conda_forge_output_validation: true
 build_platform:
   osx_arm64: osx_64
-test_on_native_only: true
+test: native
```
Comment (beckermr) [inline conda-forge.yml:4]:
```suggestion
test: native_and_emulated
```

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Pull request merged**